### ISSUE TITLE
Recent PW versions cause no results in Lister Pro when filters are empty

### ISSUE THREAD

**Issue by Toutouwai** [processwire/processwire-issues#1803]
### Short description of the issue

I realise that Lister Pro is not a core module, but this issue seems to be caused by some change in the core as the installed Lister Pro version (v1.1.5) hasn't changed.

Lister Pro instance config (`initSelector` matches 194 pages, default filter is "title contains all partial words"):

![2023-08-25_211416](https://github.com/processwire/processwire-issues/assets/1538852/6bf16e1b-885b-4648-9f68-6df9d3849c78)

PW 3.0.218 (working)

![2023-08-25_211721](https://github.com/processwire/processwire-issues/assets/1538852/b81bae84-da1b-4de0-8c1b-a9c7b0391b14)

Selector used does not include title filter (because that filter is empty):

![2023-08-25_211734](https://github.com/processwire/processwire-issues/assets/1538852/b6e04ee0-f4a0-4256-a99a-8a27e77b58c0)

PW 3.0.225 (not working) - note that selector includes empty title filter.

![2023-08-25_211529](https://github.com/processwire/processwire-issues/assets/1538852/7acd47d6-945c-4039-a1d5-87263b4200fc)

### Expected behavior

Lister Pro instance finds all pages defined in `initSelector` when filters are empty.

### Actual behavior

Lister Pro instance finds no results.

### Setup/Environment

* ProcessWire version: 3.0.225

**Comment by ryancramerdesign** [comment]
Thanks @Toutouwai I've pushed a fix for this. It looks like the issue is that you are using one of the newer operators, and Lister didn't yet recognize those operators when cleaning blank items out of the selector. So this issue has likely been here as long as those new operators like `~*=` have been around.

**Comment by Toutouwai** [comment]
> So this issue has likely been here as long as those new operators like ~*= have been around.

No, it was working with the new operators as recently as PW 3.0.218 as shown in the screenshots above.

I traced it back to this commit: https://github.com/processwire/processwire/commit/206cd8c280a9ecfb17e4311aafc95aecd7eccf7c

If you think that needs revisiting I noticed that if `$this->sessionSet("selector", $isSelector);` is always applied (as it was previous to that the commit) and only the page number reset is conditional then it resolves both the[ earlier issue](https://github.com/processwire/processwire-issues/issues/1780) and this issue. But probably the changes you have made in the latest commit are the right way to resolve this in any case.
